src/osd.c

```diff
@@ -268,6 +271,47 @@ void modeset_paint_buffer(struct modeset_buf *buf) {
 		minutes = 0;
 	}
 
+	//display custom message
+	if (osd_custom_message) {
+		FILE *file = fopen("/run/pixelpilot.msg", "r");
+		if (file != NULL) {
+
+			if (fgets(custom_msg, sizeof(custom_msg), file) == NULL) {
+				perror("Error reading from file");
+				fclose(file);
+			}
+			fclose(file);
+			if (unlink("/run/pixelpilot.msg") != 0) {
+				perror("Error deleting the file");
+			}
+			custom_msg_refresh_count = 1;
+		}
+		if (custom_msg_refresh_count > 0) {
+
+			if (custom_msg_refresh_count++ > 5) custom_msg_refresh_count=0;
+
+			size_t msg_length = strlen(custom_msg);
+
+			//remove any trailing newline that fgets may read
+			if (msg_length > 0 && custom_msg[msg_length - 1] == '\n') {
+				custom_msg[msg_length - 1] = '\0';
+				msg_length--;  // Adjust length after removing newline
+			}
```

Comment:
should this be a while loop instead? Smth like
```
while (msg_length > 0 && custom_msg[msg_length - 1] == '\n') {
				custom_msg[msg_length - 1] = '\0';
				msg_length--;  // Adjust length after removing newline
}
```

Reply:
Plan was to ensure that the read message is null terminated at position 80.
We only read `sizeof(custom_msg)` so either custom_msg[80] (and maybe before) is aleardy null terminted or the read input is exactly 80 bytes long. In that case we should always null termintare at 80.
Will change this to always null termiate at 80.

Reply:
@henkwiedig I rather meant that it could be that the file contains

```
my msg\n
\n
\n
```

Wuite unlikely, but since we already decide to strip newlines at the end of the line we read, might make sense to ensure all the newlines are stripped. But up to you. Just a tip.